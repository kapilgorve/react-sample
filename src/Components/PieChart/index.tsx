import React, { Component } from 'react';

import { Pie } from 'react-chartjs-2';
import styled from 'styled-components';

const ChartContainer = styled.div`
  display: flex;
  justify-content: center;
`;

interface PieChartProps {
  interest: number;
  totalPayable: number;
}

class PieChart extends Component<PieChartProps> {
  calculatePercentage = () => {
    const { totalPayable, interest } = this.props;
    let loanPercent: number = (interest / totalPayable) * 100;
    let interesetPercent = 100 - loanPercent;
    return [interesetPercent.toFixed(1), loanPercent.toFixed(1)];
  };

  render() {
    this.calculatePercentage();
    const options = {
      labels: ['Principla Loan Amount %', 'Total Interest %'],
      datasets: [
        {
          backgroundColor: ['#00A6B4', '#C9DE00'],
          hoverBackgroundColor: ['#003350', '#4B5000'],
          data: this.calculatePercentage(),
        },
      ],
    };
    return (
      <ChartContainer>
        <div>
          <Pie data={options} height={300} />
        </div>
      </ChartContainer>
    );
  }
}

export default PieChart;
