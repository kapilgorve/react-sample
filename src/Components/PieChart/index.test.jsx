import React from 'react';
import { cleanup, render } from '@testing-library/react';

import PieChart from './index';

afterEach(cleanup);

describe('<PieChart/> Component', () => {
  it('renders <PieChart/> ', () => {
    const { asFragment } = render(<PieChart interest="10" totalPayable="10550" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
