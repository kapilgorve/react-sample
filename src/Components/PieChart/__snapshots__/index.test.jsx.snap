// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PieChart/> Component renders <PieChart/>  1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm gMyFWH"
  >
    <div>
      <canvas
        class="chartjs-render-monitor"
        height="0"
        style="display: block;"
        width="0"
      />
    </div>
  </div>
</DocumentFragment>
`;
